src/NumericalAlgorithms/Interpolation/Actions/ElementInitInterpPoints.hpp

```diff
@@ -0,0 +1,75 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <tuple>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "NumericalAlgorithms/Interpolation/Tags.hpp"
+#include "Utilities/Requires.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+/// \cond
+namespace Parallel {
+template <typename Metavariables>
+class ConstGlobalCache;
+}  // namespace Parallel
+/// \endcond
+
+namespace intrp {
+namespace Actions {
+
+/// \ingroup ActionsGroup
+/// \brief Adds interpolation point holders to the Element's DataBox.
+///
+/// This is for the case in which the points are time-independent.
+///
+/// This should be placed in the Initialization PDAL for DgElementArray.
+///
+/// Uses: nothing
+///
+/// DataBox changes:
+/// - Adds:
+///   - `intrp::Tags::InterpPointInfo<Metavariables>`
+/// - Removes: nothing
+/// - Modifies: nothing
+struct ElementInitInterpPoints {
+  template <typename DbTags, typename... InboxTags, typename ArrayIndex,
+            typename ActionList, typename Metavariables,
+            typename ParallelComponent,
+            Requires<not tmpl::list_contains_v<
+                DbTags, intrp::Tags::InterpPointInfo<Metavariables>>> = nullptr>
+  static auto apply(db::DataBox<DbTags>& box,
+                    const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+                    const Parallel::ConstGlobalCache<Metavariables>& /*cache*/,
+                    const ArrayIndex& /*array_index*/,
+                    const ActionList /*meta*/,
+                    const ParallelComponent* const /*meta*/) noexcept {
+    return std::make_tuple(
+        db::create_from<
+            db::RemoveTags<>,
+            db::AddSimpleTags<intrp::Tags::InterpPointInfo<Metavariables>>>(
+            std::move(box),
+            db::item_type<intrp::Tags::InterpPointInfo<Metavariables>>{}));
```

Comment:
use the known type

Reply:
In this case the type is `tuples::tagged_tuple_from_typelist<
      db::wrap_tags_in<point_info_detail::WrappedPointInfoTag,
                       typename Metavariables::interpolation_target_tags,
                       Metavariables>>` which has a lot of stuff going on in it.  Should I still use the explicit type here?  (I guess I'm not sure about under what circumstances it is ok to use db::item_type).

Reply:
My vote would be for a type alias to the long thing, and then using the type alias inside. We're trying to avoid `item_type`-style code because it makes the code as difficult to read as using `auto` everywhere (I find, at least).